lib/teracy-dev/config/manager.rb

```diff
@@ -8,25 +8,48 @@ class Manager
 
       def initialize
         if !!@@instance
-          raise "TeracyDev::Processors::Manager can only be initialized once"
+          raise "TeracyDev::Config::Manager can only be initialized once"
         end
         @@instance = self
 
         @logger = TeracyDev::Logging.logger_for(self.class.name)
         @configurators = []
       end
 
-      def register(configurator)
+      def register(configurator, weight)
         if !configurator.respond_to?(:configure)
           @logger.warn("configurator #{configurator} must implement configure method, ignored")
           return
         end
-        @configurators << configurator
+
+        unless weight.is_a? Integer and (0..9).include?(weight)
+          @logger.warn("#{configurator}'s weight must be integer and have value in range 0.. 9, otherwise weight will be set to default (5)")
+          weight = 5
+        end
+
+        @configurators << { configurator: configurator, weight: weight }
```

Comment:
try this: https://stackoverflow.com/questions/28234803/ruby-sort-by-multiple-fields-and-multilple-directions-for-different-data-types

processors << [id: processors.length, processor: processor, weight: weight]

and to sort with our requirement:

sorted_processors = multi_sort(processors, id: :asc, weight: :desc)